Add configure options for all providers except localnet.

Patch by Kevin McCarthy <signals@gentoo.org>

--- configure.ac
+++ configure.ac
@@ -158,7 +158,71 @@
 AC_SUBST(CONNECTIVITY_LIBS)
 AC_SUBST(CONNECTIVITY_CFLAGS)
 
-PROVIDER_SUBDIRS="example hostip geonames nominatim manual plazes localnet yahoo gsmloc"
+PROVIDER_SUBDIRS="localnet"
+
+AC_ARG_ENABLE(hostip,
+              AS_HELP_STRING([--enable-hostip=@<:@no/yes/auto@:>@],
+                             [build with hostip support]), ,
+                             enable_hostip=auto)
+
+if test "x$enable_hostip" != "xno"; then
+   PROVIDER_SUBDIRS="$PROVIDER_SUBDIRS hostip"
+fi
+
+AC_ARG_ENABLE(geonames,
+              AS_HELP_STRING([--enable-geonames=@<:@no/yes/auto@:>@],
+                             [build with geonames support]), ,
+                             enable_geonames=auto)
+
+if test "x$enable_geonames" != "xno"; then
+   PROVIDER_SUBDIRS="$PROVIDER_SUBDIRS geonames"
+fi
+
+AC_ARG_ENABLE(nominatim,
+              AS_HELP_STRING([--enable-nominatim=@<:@no/yes/auto@:>@],
+                             [build with nominatim support]), ,
+                             enable_nominatim=auto)
+
+if test "x$enable_nominatim" != "xno"; then
+   PROVIDER_SUBDIRS="$PROVIDER_SUBDIRS nominatim"
+fi
+
+AC_ARG_ENABLE(manual,
+              AS_HELP_STRING([--enable-manual=@<:@no/yes/auto@:>@],
+                             [build with manual support]), ,
+                             enable_manual=auto)
+
+if test "x$enable_manual" != "xno"; then
+   PROVIDER_SUBDIRS="$PROVIDER_SUBDIRS manual"
+fi
+
+AC_ARG_ENABLE(plazes,
+              AS_HELP_STRING([--enable-plazes=@<:@no/yes/auto@:>@],
+                             [build with plazes support]), ,
+                             enable_plazes=auto)
+
+if test "x$enable_plazes" != "xno"; then
+   PROVIDER_SUBDIRS="$PROVIDER_SUBDIRS plazes"
+fi
+
+AC_ARG_ENABLE(yahoo,
+              AS_HELP_STRING([--enable-yahoo=@<:@no/yes/auto@:>@],
+                             [build with yahoo support]), ,
+                             enable_yahoo=auto)
+
+if test "x$enable_yahoo" != "xno"; then
+   PROVIDER_SUBDIRS="$PROVIDER_SUBDIRS yahoo"
+fi
+
+AC_ARG_ENABLE(gsmloc,
+              AS_HELP_STRING([--enable-gsmloc=@<:@no/yes/auto@:>@],
+                             [build with gsmloc support]), ,
+                             enable_gsmloc=auto)
+
+if test "x$enable_gsmloc" != "xno"; then
+   PROVIDER_SUBDIRS="$PROVIDER_SUBDIRS gsmloc"
+fi
+
 
 # -----------------------------------------------------------
 # gypsy / gpsd / skyhook
